robin.js

```diff
@@ -16,11 +16,13 @@ module.exports = function (n, ps) {  // n = num players
     ps.push(DUMMY); // so we can match algorithm for even numbers
     n += 1;
   }
-  for (var j = 0; j < n - 1; j += 1) {
+  for (let j = 0; j < n - 1; j += 1) {
     rs[j] = []; // create inner match array for round j
-    for (var i = 0; i < n / 2; i += 1) {
-      if (ps[i] !== DUMMY && ps[n - 1 - i] !== DUMMY) {
-        rs[j].push([ps[i], ps[n - 1 - i]]); // insert pair as a match
+    for (let i = 0; i < n / 2; i += 1) {
+      const o = n - 1 - i;
+      if (ps[i] !== DUMMY && ps[o] !== DUMMY) {
+        const iHome = i === 0 && j % 2 === 1;
```

Comment:
is this really enough to classify it? it looks like you're only flipping the first match aorund every other round?

Reply:
Flipping the first match around every other round was indeed what I intended to do. 

I tested it with 3,4,5,6 and 7 teams and got the expected outcome as described in my [original comment](https://github.com/clux/roundrobin/pull/3#issue-676537847) for each amount. 

I can test with an even larger number of teams in a bit if you'd like.

Reply:
Ok. Yeah, I think it actually would be helpful to have a few more test cases as assurance here. Maybe one odd and one even  case (say 9 and 10?), and check that the number of home/away matches are what is expected for all the players.

Reply:
Test with 9:
```js
robin(9);
[
  [ [ 2, 9 ], [ 3, 8 ], [ 4, 7 ], [ 5, 6 ] ],
  [ [ 9, 1 ], [ 2, 7 ], [ 3, 6 ], [ 4, 5 ] ],
  [ [ 1, 8 ], [ 9, 7 ], [ 2, 5 ], [ 3, 4 ] ],
  [ [ 7, 1 ], [ 8, 6 ], [ 9, 5 ], [ 2, 3 ] ],
  [ [ 1, 6 ], [ 7, 5 ], [ 8, 4 ], [ 9, 3 ] ],
  [ [ 5, 1 ], [ 6, 4 ], [ 7, 3 ], [ 8, 2 ] ],
  [ [ 1, 4 ], [ 5, 3 ], [ 6, 2 ], [ 8, 9 ] ],
  [ [ 3, 1 ], [ 4, 2 ], [ 6, 9 ], [ 7, 8 ] ],
  [ [ 1, 2 ], [ 4, 9 ], [ 5, 8 ], [ 6, 7 ] ]
]
```
**all** teams play **4 away** and **4 home** matches.
<br />
Test with 10:
```js
robin(10);
[
  [ [ 1, 10 ], [ 2, 9 ], [ 3, 8 ], [ 4, 7 ], [ 5, 6 ] ],
  [ [ 9, 1 ], [ 10, 8 ], [ 2, 7 ], [ 3, 6 ], [ 4, 5 ] ],
  [ [ 1, 8 ], [ 9, 7 ], [ 10, 6 ], [ 2, 5 ], [ 3, 4 ] ],
  [ [ 7, 1 ], [ 8, 6 ], [ 9, 5 ], [ 10, 4 ], [ 2, 3 ] ],
  [ [ 1, 6 ], [ 7, 5 ], [ 8, 4 ], [ 9, 3 ], [ 10, 2 ] ],
  [ [ 5, 1 ], [ 6, 4 ], [ 7, 3 ], [ 8, 2 ], [ 9, 10 ] ],
  [ [ 1, 4 ], [ 5, 3 ], [ 6, 2 ], [ 7, 10 ], [ 8, 9 ] ],
  [ [ 3, 1 ], [ 4, 2 ], [ 5, 10 ], [ 6, 9 ], [ 7, 8 ] ],
  [ [ 1, 2 ], [ 3, 10 ], [ 4, 9 ], [ 5, 8 ], [ 6, 7 ] ]
]
```
teams **1,3,5,7,9** play **5 away** and **4 home** matches.
teams **2,4,6,8,10** play **4 away** and **5 home** matches.
<br />
(note: away = left, home = right)
<br />
Everything is working as expected!

Reply:
Excellent :-)

Do you mind putting those in a `home-away` test case? If not, don't worry. I am happy to merge as is.

Was thinking something like a `t.deepEquals(robin(10, youroutput)` plus some kind home/away count like like:
```
homecount = Array(n).fill(0)
awaycount = Array(n).fill(0)
for r in rounds {
  for pair in r {
    // match up homecounts for player s
    homecount[pair[0]-1] + 1;
    awaycount[pair[1]-1] + 1
  }
}
// verify numbers are either equal in odd case, or at most one of in even case
```

Reply:
I will look into doing this in the morning!

Reply:
Finished writing the tests and pushed in commit 2e8949c871b4d43e525bdefcdcbeb39552a1484d
```js
test('home-away', function *(t) {
  function hasCorrectHomeAwayOutput(n) {
    const awayPlayers = [];
    const homePlayers = [];

    for (const round of robin(n)) {
      for (const match of round) {
        awayPlayers.push(match[0]);
        homePlayers.push(match[1]);
      }
    }

    for (let i = 1; i <= n; i++) {
      const isOddPlayer = i % 2 === 1;
      const fairAmount = (n - 1) / 2;
      const [ roundedUp, roundedDown ] = [ Math.ceil(fairAmount), Math.floor(fairAmount) ];
      const [ awayAmount, homeAmount ] = isOddPlayer ? [ roundedUp, roundedDown ] : [ roundedDown, roundedUp ];
      t.eq(
        awayPlayers.filter(player => player === i).length,
        awayAmount,
        `player ${i} of ${n} plays the correct amount of away matches (${awayAmount})`
      );
      t.eq(
        homePlayers.filter(player => player === i).length,
        homeAmount,
        `player ${i} of ${n} plays the correct amount of home matches (${homeAmount})`
      );
    }
  }

  hasCorrectHomeAwayOutput(9);
  hasCorrectHomeAwayOutput(10);
})
```
As you can see I pushed tests for 9 and 10 players, but I also tested 1000 players locally and it worked!

![image](https://user-images.githubusercontent.com/37840393/123298764-d1a8cf80-d4e6-11eb-8ad1-e26f946cc236.png)

After seeing these test results I think it's 100% safe to say that these changes do in fact classify what we want to accomplish and does scale lol :)

Reply:
Hah, that's amazing. And a better way of doing it than i suggested (with indexes) too. Thank you so much. I'll make a release later on from this!